### PR TITLE
[kubeadm] cleanup deps on pkg/version

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Refactor k8s.io/kubernetes/pkg/version -> k8s.io/client-go/pkg/version

**Which issue(s) this PR fixes**:

Ref kubernetes/kubeadm#1600

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by yagonobre — COMMENTED** [review]
@SataQiu please batch the changes, so we can fix this issue with few prs.

**Comment by SataQiu** [comment]
/hold
@yagonobre 
Thanks! I will try to add more changes.

File: cmd/kubeadm/.import-restrictions [hunk 1]
```diff
@@ -89,7 +89,7 @@
 				"k8s.io/kubernetes/pkg/util/procfs",
 				"k8s.io/kubernetes/pkg/util/sysctl",
 				"k8s.io/kubernetes/pkg/util/taints",
-				"k8s.io/kubernetes/pkg/version"
+				"k8s.io/client-go/pkg/version"
```
Comment (neolit123) [inline cmd/kubeadm/.import-restrictions]: there is already a rule about client-go near the top of the file.

**Review by rosti — APPROVED** [review]
Thanks @SataQiu !
I think, that dropping off the `//pkg/version` dependency is good enough for a single PR.
Let's however fix the last dependency in `./cmd/kubeadm/BUILD`. We should probably do what kubectl did and have a local copy of `version_x_defs`.

File: cmd/kubeadm/app/version/base.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	// TODO: This field is still called "gitVersion" for legacy
+	// reasons. For prerelease versions, the build metadata on the
+	// semantic version is a git hash, but the version itself is no
+	// longer the direct output of "git describe", but a slight
+	// translation to be semver compliant.
+
+	// NOTE: The $Format strings are replaced during 'git archive' thanks to the
+	// companion .gitattributes file containing 'export-subst' in this same
+	// directory.  See also https://git-scm.com/docs/gitattributes
+	gitVersion   = "v0.0.0-master+$Format:%h$"
```
Comment (neolit123) [inline cmd/kubeadm/app/version/base.go]: /hold
i'm not convinced we need a local copy of a format that is already defined in client-go.

**Review by neolit123 — CHANGES_REQUESTED** [review]

**Comment by SataQiu** [comment]
@neolit123 What puzzles me is how do we resolve the dependency on pkg/version for cmd/kubeadm/BUILD

**Comment by SataQiu** [comment]
@neolit123  Updated!

**Review by neolit123 — COMMENTED** [review]
/hold cancel
thanks. LGTM this is much better.

final `/lgtm` on @rosti
/assign @rosti

**Comment by SataQiu** [comment]
@rosti Could you please look at it again?

**Review by rosti — APPROVED** [review]
Thanks @SataQiu !
/lgtm

**Pull request merged**